Swift/Database.swift

```diff
@@ -515,14 +470,8 @@ public final class Database {
     
     /// Get the default collection. If the default collection is deleted, nil will be returned.
     public func defaultCollection() throws  -> Collection? {
-        do {
-            let _ = try _impl.defaultCollection()
-        } catch let err as NSError {
-            _defaultCollection = nil
-            throw err
-        }
-        
-        return _defaultCollection
+        let c = try _impl.defaultCollection()
+        return Collection(c, db: self)
```

Comment:
It would be nice if we can cache the default collection but that will cause the circular reference that will require to bring back the weak database in Collection and Scope. Let's do this at least for now.